opendevin/runtime/utils/runtime_build.py

```diff
@@ -60,44 +71,64 @@ def _generate_dockerfile(
     if skip_init:
         dockerfile_content = f'FROM {base_image}\n'
     else:
+        # Ubuntu 22.x has libgl1-mesa-glx, but 24.x and above have libgl1!
+        if 'ubuntu' in base_image and (
+            base_image.endswith(':latest') or base_image.endswith(':24.04')
+        ):
+            LIBGL_MESA = 'libgl1'
+        else:
+            LIBGL_MESA = 'libgl1-mesa-glx'
+
```

Comment:
Would add this at the top:
```py
        dockerfile_content = (
            'ENV DEBIAN_FRONTEND=noninteractive '
            'POETRY_NO_INTERACTION=1 '
            'POETRY_VIRTUALENVS_IN_PROJECT=1 '
            'POETRY_VIRTUALENVS_CREATE=1 '
            'POETRY_CACHE_DIR=/tmp/poetry_cache\n'
        )
```
(and "+=" in the below line then)

Reply:
Is there any particular reason why we should add this when not adding it still works? I'm trying to see if we can make things as simple as possible..

Reply:
They're also in the app dockerfile, thought they'd be needed:
https://github.com/OpenDevin/OpenDevin/blob/f3c23e80395bbe95fb8935ff1f8b2b6e563e3a5f/containers/app/Dockerfile#L18

Reply:
🤔 https://python-poetry.org/docs/configuration/#virtualenvsin-project

> If not set explicitly, poetry by default will create a virtual environment under {cache-dir}/virtualenvs or use the {project-dir}/.venv directory if one already exists.

Now i see why the previous runs fails, because after `rm -rf /root/.cache/pypoetry` the virtual env will be removed.

I don't think we need this for the sandbox, because we will `rm -rf /opendevin/code` and copy a new version of the codebase there, if we enable `POETRY_VIRTUALENVS_IN_PROJECT`, it will delete the virtual env with `rm -rf /opendevin/code` , and we will have to re-install everything from scratch whenever we update the source code..

Reply:
And `POETRY_VIRTUALENVS_CREATE` also default to true.. https://python-poetry.org/docs/configuration/#virtualenvscreate